src/pykeen/sampling/negative_sampler.py

```diff
@@ -44,7 +53,27 @@ def num_entities(self) -> int:  # noqa: D401
         """The number of entities to sample from."""
         return self.triples_factory.num_entities
 
+    @property
+    def num_relations(self) -> int:  # noqa: D401
+        """The number of relations to sample from."""
+        return self.triples_factory.num_relations
+
     @abstractmethod
     def sample(self, positive_batch: torch.LongTensor) -> torch.LongTensor:
         """Generate negative samples from the positive batch."""
         raise NotImplementedError
+
+    def _filter_negative_triples(self, negative_batch: torch.LongTensor) -> torch.LongTensor:
```

Comment:
is this necessary? I thought we argued in the paper that we could skip this because it's low probability

At least make a note in the docstring on when this assumption isn't so good

Reply:
That depends. In usual datasets it's low probability. However there are some aspects to it.
E.g. is this expected to act a bit as a regularization, since it adds noise signal to the training data. To make things more complicated, the added noise signal depends on the ratio of true triples for a given entity relation or entity entity pair. Therefore, the effects are hard to distinguish and a researcher might want to exclude the possibility of having false negatives in the proposed negative triples.

Reply:
sounds like something that could go in the module docstring ;)